frontend/ui/widget/menu.lua

```diff
@@ -1176,7 +1188,71 @@ function Menu:getPageNumber(item_number)
     if #self.item_table == 0 or item_number == 0 then
         return 1
     end
-    return math.ceil(math.min(item_number, #self.item_table) / self.perpage)
+    if self.items_max_lines then
+        for page, items in ipairs(self.page_items) do
+            if item_number <= items[#items] then
+                return page
+            end
+        end
+        return #self.page_items
+    else
+        return math.ceil(math.min(item_number, #self.item_table) / self.perpage)
+    end
+end
+
+function Menu:setupItemHeights()
```

Comment:
(Question for English scholars : each item may have its own height. Should this be named setupItemHeights (meaning, the multiple heights of a single item), or setupItem**s**Height (all/each item's height) or setupItem**s**Height**s** ?)

Reply:
>setupItemHeights

This one.

Reply:
Yeah, as it is. ^_^